crictl e2e test is failing.

It is currently blocking PRs.
```
Summarizing 8 Failures:
[Fail] crictl: pull [It] should succeed without tag or digest 
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/test/framework/framework.go:104
[Fail] crictl: pull [It] should succeed with tag 
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/test/framework/framework.go:104
[Fail] crictl: pull [It] should fail on not existing image 
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/test/framework/framework.go:126
[Fail] crictl: info [It] should succeed 
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/test/framework/framework.go:104
[Fail] crictl: info [AfterEach] should fail with additional argument 
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/test/framework/framework.go:205
[Fail] crictl: pull [It] should succeed with digest 
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/test/framework/framework.go:104
[Fail] crictl: pull [AfterEach] should succeed to show the help 
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/test/framework/framework.go:205
[Fail] crictl: version [It] should succeed 
/home/travis/gopath/src/github.com/kubernetes-sigs/cri-tools/test/framework/framework.go:104
Ran 12 of 12 Specs in 246.295 seconds
FAIL! -- 4 Passed | 8 Failed | 0 Pending | 0 Skipped
Ginkgo ran 1 suite in 4m18.491441555s
Test Suite Failed
Makefile:108: recipe for target 'test-e2e' failed
make: *** [test-e2e] Error 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -73,8 +73,6 @@ jobs:
       name: crictl e2e
       os: linux
       script:
-        # TODO: Re-enable the test when it is fixed.
-        - exit 0
         - |
           sudo apt-get update &&\
           sudo apt-get install -y libseccomp-dev

--- a/test/framework/framework.go
+++ b/test/framework/framework.go
@@ -32,6 +32,8 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+const timeout = 10 * time.Minute
+
 // TestFramework is used to support commonly used test features
 type TestFramework struct {
 	crioDir string
@@ -128,18 +130,37 @@ func (t *TestFramework) CrictlExpectFailureWithEndpoint(
 
 func SetupCrio() string {
 	const (
-		crioURL = "https://github.com/cri-o/cri-o"
-		timeout = 10 * time.Minute
+		// TODO: update to 1.16.0 once released
+		crioURL       = "https://github.com/cri-o/cri-o"
+		crioVersion   = "aabfe2eb73c92dc2378b36f0fc782033d9460fda"
+		conmonURL     = "https://github.com/containers/conmon"
+		conmonVersion = "v2.0.1"
 	)
 	tmpDir := filepath.Join(os.TempDir(), "crio-tmp")
 
 	if _, err := os.Stat(tmpDir); os.IsNotExist(err) {
 		logrus.Info("cloning and building CRI-O")
-		lcmd("git clone --depth=1 %s %s", crioURL, tmpDir).Wait(timeout)
-		cmd(tmpDir, "make").Wait(timeout)
+
+		Expect(checkoutAndBuild(tmpDir, crioURL, crioVersion)).To(BeNil())
+
+		conmonTmp := filepath.Join(tmpDir, "conmon")
+		checkoutAndBuild(conmonTmp, conmonURL, conmonVersion)
 	}
 
 	return tmpDir
+}
+
+func checkoutAndBuild(dir, url, rev string) error {
+	// A much faster approach than just cloning the whole repository
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		return err
+	}
+	cmd(dir, "git init").Wait(timeout)
+	cmd(dir, "git remote add origin %s", url).Wait(timeout)
+	cmd(dir, "git fetch --depth=1 origin %s", rev).Wait(timeout)
+	cmd(dir, "git checkout -f FETCH_HEAD").Wait(timeout)
+	cmd(dir, "make").Wait(timeout)
+	return nil
 }
 
 // Start the container runtime process
@@ -179,7 +200,7 @@ func (t *TestFramework) StartCrio() (string, string, *Session) {
 		" --storage-driver=vfs",
 		filepath.Join(tmpDir, "bin", "crio"),
 		endpoint,
-		filepath.Join(tmpDir, "bin", "conmon"),
+		filepath.Join(t.crioDir, "conmon", "bin", "conmon"),
 		filepath.Join(tmpDir, "exits"),
 		filepath.Join(tmpDir, "attach"),
 		filepath.Join(tmpDir, "log"),

